cmd/remote-storage/app/server_test.go

```diff
@@ -133,40 +135,40 @@ func TestNewServer_TLSConfigError(t *testing.T) {
 				TLSSetting: tlsCfg,
 			},
 		},
-		storageMocks.factory,
+		f,
 		tenancy.NewManager(&tenancy.Options{}),
 		telset,
 	)
 	assert.ErrorContains(t, err, "failed to load TLS config")
 }
 
-func TestCreateGRPCHandler(t *testing.T) {
-	storageMocks := newStorageMocks()
-	h, err := createGRPCHandler(storageMocks.factory, zap.NewNop())
-	require.NoError(t, err)
+// func TestCreateGRPCHandler(t *testing.T) {
```

Comment:
@yurishkuro What're your thoughts on this test? It's a bit challenging now to recreate this test since we pass in a real factory so we would need to look at the implementations of the actual factories to try and induce errors. Do you think there's value in that?

Reply:
you can change `createGRPCHandler` to accept an interface (even locally defined) instead of a struct, then it can be mocked.